### PR TITLE
Add utility functions

### PR DESCRIPTION
**Pull request by lavoiesl**
Add `Exists`, `Read`, and `Write`, which are common operations.

### CODE REVIEW

File: pkg/utils.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	var file *File
+	if file, err = fs.Open(ctx, path, options); err != nil {
+		return nil, fmt.Errorf("unable to open %s: %w", path, err)
+	}
+
+	if data, err = ioutil.ReadAll(file); err != nil {
+		return nil, fmt.Errorf("unable to read %s: %w", path, err)
+	}
+
+	if err = file.Close(); err != nil {
```
Comment (bmassemin) [inline pkg/utils.go]: Should we use `defer` for closing the file after opening it ?

File: pkg/errors.go [hunk 2]
```diff
@@ -16,8 +16,11 @@ type isNotExister interface {
 // IsNotExist returns a boolean indicating whether the error is known to report that
 // a path does not exist.
 func IsNotExist(err error) bool {
-	e, ok := err.(isNotExister)
-	return ok && e.isNotExist()
+	var e isNotExister
```
Comment (Emile-Filteau) [inline pkg/errors.go]: Why the change here?

File: pkg/utils.go (hunk 1 above)
Comment (lavoiesl) [reply pkg/utils.go]: Maybe not `defer`, because I want the error to bubble up, but you're right that I should add it before the return of unable to read.

File: pkg/errors.go (hunk 2 above)
Comment (lavoiesl) [reply pkg/errors.go]: Because of the use of `fmt.Errorf`, the error is now wrapped. I want `IsNotExist` to still be able to recognized a wrapped error.

File: pkg/utils.go (hunk 1 above)
Comment (pior) [reply pkg/utils.go]: Interesting post on the subject: https://www.joeshaw.org/dont-defer-close-on-writable-files/

Comment (pior) [reply pkg/utils.go]: I also usually lean toward `defer`. But for this simple function, this is cleaner/simpler.
